assembl/models/langstrings.py

```diff
@@ -464,6 +464,49 @@ def EMPTY(cls, db=None):
     def reset_cache(cls):
         pass
 
+    # Which object owns this?
+    owner_object = None
+    _owning_relns = []
+
+    @classmethod
+    def setup_ownership_load_event(cls, owner_class, relns):
+        def load_owner_object(target, context):
+            for reln in relns:
+                if reln in target.__dict__:
+                    getattr(target, reln).owner_object = target
+        event.listen(owner_class, "load", load_owner_object, propagate=True)
+        event.listens_for(owner_class, "refresh", load_owner_object, propagate=True)
+        def set_owner_object(target, value, old_value, initiator):
+            if old_value is not None:
+                old_value.owner_object = None
+            value.owner_object = target
+        for reln in relns:
+            cls._owning_relns.append((owner_class, reln))
+            event.listen(getattr(owner_class, reln), "set", set_owner_object, propagate=True)
```

Comment:
Yes, we recalculate ownership when the langstring is set on the name (eg.) Not sure I understand your second question, but the relationship itself has no crud.